eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -110,8 +110,67 @@ stages:
             Exit 1
           }
 
+          $checkoutCommitish = '${{ parameters.SdkRepoCommit }}'
+          Write-Host "##vso[task.setvariable variable=CheckoutCommitish]$checkoutCommitish"
+          Write-Host "Will use commitish for SDK repo checkout: '$checkoutCommitish'"
         displayName: "Create Run Time Variables"
 
+      - ${{ if and(eq(variables['System.TeamProject'], 'internal'), eq(parameters.CreatePullRequest, true), ne(variables['Build.Reason'], 'PullRequest')) }}:
+        - pwsh: |
+            $sdkRepoCommitish = '${{ parameters.SdkRepoCommit }}'
+            $sdkRepoUrl = '${{ parameters.SdkRepoUrl }}'
+            Write-Host "Validating SDK repo commitish: '$sdkRepoCommitish'"
+            # Check if it's a commit SHA
+            $commitShaPattern = '^[a-fA-F0-9]{7,40}$'
+
+            if ($sdkRepoCommitish.contains(":")) {
+              Write-Host "##vso[task.logissue type=error]'SDK repository commit' must be a commit SHA or branch name without ':'"
+              Exit 1
+            }
+
+            if (($sdkRepoCommitish -ne 'main') -and ($sdkRepoCommitish -notmatch $commitShaPattern)) {
+                # set repo owner to 'azure-sdk' when customized branch is provided
+                $sdkRepoOwner = 'azure-sdk'
+                Write-Host "##vso[task.setvariable variable=SdkRepoOwner]$sdkRepoOwner"
+                Write-Host "SdkRepoOwner variable set to: $sdkRepoOwner"
+                $sdkRepoUrl = "https://github.com/azure-sdk/$(SdkRepoName)"
+                # Check if branch exists remotely
+                Write-Host "Checking if branch '$sdkRepoCommitish' exists in remote repository '$sdkRepoUrl'..."
+
+                $branchExists = git ls-remote --heads $sdkRepoUrl "refs/heads/$sdkRepoCommitish"
+                $gitExitCode = $LASTEXITCODE
+
+                if ($gitExitCode -ne 0) {
+                    Write-Host "##vso[task.logissue type=error]Failed to validate branch remotely (git exit code: $gitExitCode)"
+                    Exit $gitExitCode
+                }
+
+                if ([string]::IsNullOrEmpty($branchExists)) {
+                    Write-Host "Branch '$sdkRepoCommitish' does not exist in remote repository"
+                    Write-Host "Will checkout 'main' and create new branch '$sdkRepoCommitish' later"
+
+                    # Set flags for post-checkout processing
+                    Write-Host "##vso[task.setvariable variable=CreateNewBranch]true"
```

Comment:
I don't think we need this. All we need is to set the `CheckoutCommitish` to either be main or the remote branch it exists. Then you need to always set your `SdkPullRequestSourceBranch` to either what was passed in if or the generated one if it isn't main. I think you can eliminate the other 2 steps you added below.